Source/RxSwift/toggleBool.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  toggleBool.swift
+//  RxSwiftExt-iOS
+//
+//  Created by Vincent Pradeilles on 20/02/2018.
+//  Copyright © 2018 RxSwiftCommunity. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+extension ObservableType {
+    
+    /**
+     Returns an observable sequence of booleans, starting with `initial` and toggling its value
+     at each `.next` event
+     
+     - parameter initial: the boolean value to initialize the sequence
+     - returns: An observable sequence of booleans that starts with `initial` and toogles its value at each `.next` event
+     */
+    public func toggleBool(initial: Bool) -> Observable<Bool> {
+        return scan(initial) { previous, _ in return !previous }.startWith(initial)
```

Comment:
I think we can change this to `reduce(into: initial) { previous, _ in previous.toggle() }`

Thoughts?

Reply:
`reduce` emits only final value. Are you sure such behavior is required?

Reply:
meant scan into.

Reply:
It doesn't emit initial value

Reply:
Actually, I just wonder why `startWith` needed at all. I don't want sequence to emit something before source

Reply:
I think you're missing my point. I wanted to use scan(into) and `toggle`. The `startWith` should still be there.

Reply:
I mean common example will be like:
```
button.rx.tap.toggleBool(initial: true).bind(to: view.rx.isHidden)
```
And i want view becomes hidden or not only after first button tap

Reply:
> use scan(into) and toggle
What's the difference with the current? Quite subjective IMO. But yes, can be implemented in that way.

Reply:
Is that all? I can create new PR with that change

Reply:
I just think this is is mirroring Foundation's toggle, so it's nicer to use that. Yes, it's quite subjective.

Reply:
TBH I can't really say this operator had too much community interest. Evidently only a few people (2-3) have asked for it and it's really minor. Don't mind adding it since it's in Foundation but I think it's quite minor anyways.

Reply:
If it's not popular/needed - it's better to close it.